shaky/src/main/java/com/linkedin/android/shaky/FormFragment.java

```diff
@@ -133,20 +178,80 @@ public boolean onMenuItemClick(MenuItem item) {
     /**
      * Validates the message and returns true if the form is valid.
      */
-    private boolean validate(@NonNull String message) {
+    private boolean validateMessage(@NonNull String message) {
         if (message.trim().length() == 0) {
-            AlertDialog alertDialog = new AlertDialog.Builder(getActivity()).create();
-            alertDialog.setMessage(getString(R.string.shaky_empty_feedback_message));
-            alertDialog.setButton(AlertDialog.BUTTON_POSITIVE, getString(R.string.shaky_empty_feedback_confirm),
-                                  new DialogInterface.OnClickListener() {
-                                      public void onClick(DialogInterface dialog, int which) {
-                                          dialog.dismiss();
-                                      }
-                                  });
-            alertDialog.show();
+            showAlertDialog(getString(R.string.shaky_empty_feedback_message));
             return false;
         }
+        return true;
+    }
 
+    private boolean validateSubcategory(int selectedItemPosition) {
+        if (selectedItemPosition == 0) {
+            showAlertDialog(getString(R.string.shaky_subcategory_not_selected_message));
```

Comment:
The only difference of validate and validateMessage is checking the empty string case. we can just reuse the original validate method without creating a new helper function here. Showing an empty dialog to user is also not what we want

Reply:
Discussed offline, agreed to keep the two functions separate